mutiny-core/src/nodemanager.rs

```diff
@@ -708,6 +714,35 @@ impl<S: MutinyStorage> NodeManager<S> {
             Arc::new(Mutex::new(nodes_map))
         };
 
+        let fedimint_storage = storage.get_fedimints()?;
+
+        let unarchived_fedimints = fedimint_storage
+            .clone()
+            .fedimints
+            .into_iter()
+            .filter(|(_, n)| !n.is_archived());
+
+        let mut fedimint_map = HashMap::new();
+
+        for fedimint_item in unarchived_fedimints {
+            let fedimint = FedimintClient::new(
+                fedimint_item.0,
+                &fedimint_item.1,
+                fedimint_item.1.federation_code.clone(),
+                c.xprivkey,
+                storage.clone(),
+                c.network,
+                logger.clone(),
+            )
+            .await?;
+
+            let id = fedimint.fedimint_client.federation_id();
+
+            fedimint_map.insert(id, Arc::new(fedimint));
+        }
+
+        let fedimints = Arc::new(Mutex::new(fedimint_map));
```

Comment:
should these be in lib instead of nodemanager? Imo that'd make more sense

Reply:
moved, might still clean it up a bit in lib though.